app/services/github_handler.rb

```diff
@@ -0,0 +1,64 @@
+class GithubHandler
+  def initialize(event, payload, signature)
+    @event = event
+    @payload = JSON.parse(payload)
+    @raw_payload = payload
+    @signature = signature
+  end
+
+  def handle
+    return unless webhook_verified?
+
+    case @event
+    when "pull_request"
+      case @payload["action"]
+      when "review_requested"
+        handle_review_request
+        # when "review_request_removed"
+        # TODO: Call the method to handle a review request removal
+        # else unsupported
+      end
+      # else unsupported
+    end
+  end
+
+  def handle_review_request
+    owner = create_or_find_user(@payload["pull_request"]["user"])
+    pr = create_or_find_pr(@payload["pull_request"])
+    # Even if you select multiple reviewers at once, the webhook sends a post for every person selected
+    # we can assume it is going to be a single user obj "requested_reviewer"
+    participants = [create_or_find_user(@payload["requested_reviewer"])]
+    ReviewRequest.create(data: @payload, owner: owner, users: participants, pull_request: pr)
```

Comment:
I would use something like `pr.review_requests.create(data: @payload, owner: owner, users: participants)`

Reply:
What happens if this creation fails? (`create` vs `create!`?)